ui/Screen/Lock.svelte

```diff
@@ -6,20 +6,48 @@
  LICENSE file.
 -->
 <script lang="typescript">
-  import * as screen from "../src/screen";
-  import * as session from "../src/session";
+  import { tick as svelteTick } from "svelte";
+  import * as screen from "ui/src/screen";
+  import * as session from "ui/src/session";
+  import * as error from "ui/src/error";
+  import * as notification from "ui/src/notification";
 
   import { Button, Emoji, PasswordInput } from "ui/DesignSystem";
 
   let passphrase = "";
   let unlockInProgress = false;
+  let input: PasswordInput;
+
+  let errorNotificationHandle: notification.Handle | undefined;
 
   const unlock = async () => {
+    if (errorNotificationHandle) {
+      errorNotificationHandle.remove();
+    }
     await screen.withLock(async () => {
       unlockInProgress = true;
-      await session.unseal(passphrase).finally(() => {
+      try {
+        const unlocked = await session.unseal(passphrase);
+        if (!unlocked) {
+          passphrase = "";
+          // We wait until the component has re-rendered with
+          // `unlockInProgress = false` and the input has been enabled
+          // again.
+          svelteTick().then(() => {
```

Comment:
For my own understanding: is there a reason why we don't `await` here instead of `.then`?

Reply:
If we `await` `svelteTick`, then `unlockInProgress` will be set to `false` _after_ the svelte tick. This would defeat the purpose of using `svelteTick`.